hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -389,7 +391,7 @@
   // the key is table name, the value is the number of compactions in that table.
   private Map<TableName, AtomicInteger> mobCompactionStates = Maps.newConcurrentMap();
 
-  MasterCoprocessorHost cpHost;
+  volatile MasterCoprocessorHost cpHost;
```

Comment:
Just a casual comment here:- Volatile comes with a performance penalty because of the constraints the JVM optimizer has to follow and extra memory copy. But at the same time, I don't know if we can measure the impact yet.

Reply:
and even if we can measure the impact, this may be still required because at the moment of the some getter is using the coprocessor for reading or calculating their methods, this loaded CP(s) could be changed or removed. 

I noticed `RackManager` in `BaseLoadBalancer`/`LoadBalancer` is also having this penalty and if we don't change the configuration, we should not go into the `onConfigurationChange` hooked called by `update_all_config`.

So, IMO the benefit is better than restarting the JVM .